bot/exts/utils/snekbox.py

```diff
@@ -217,6 +217,71 @@ async def send_eval(self, ctx: Context, code: str) -> Message:
             log.info(f"{ctx.author}'s job had a return code of {results['returncode']}")
         return response
 
+    @staticmethod
+    def get_time(results: dict) -> Tuple[str, Optional[str]]:
+        """
+        Parses time and output from results dict, if there was an error time is None and output contains the error.
+
+        Returns code output and timeit results
+        """
+        output = results["stdout"].strip("\n")
+
+        if results["returncode"] == 0:
+            time_index = output.rfind("\n")
+            if time_index != -1:
+                return output[:time_index], output[time_index:]
+            return "", output
+
+        return output, None
+
+    async def send_timeit(self, ctx: Context, code: str) -> Message:
+        """
+        Evaluate code with timing, format it, and send the output to the corresponding channel.
+
+        Return the bot response.
+        """
+        async with ctx.typing():
+            code = code.replace("'", r"\'")
+            code = f"import timeit\n" \
+                   f"timeit.main(['{code}'])"
```

Comment:
I was envisioning this as a command that only returns the execution time. In other words, no output from the actual code. I'm not sure if everyone agrees with this direction.

In any case, it'd be difficult to suppress output when using `timeit.main()` since it also writes to stdout. `contextlib.redirect_stdout` doesn't really work since it's simple to bypass e.g. by writing to `/dev/stdout` directly.

Reply:
I agree, the tool would be used when wanting to compare running times, the output isn't the focus here. For that you have the normal eval. It's also more in line with how standard timing tools work.